File: src/main/java/me/ag2s/umdlib/domain/UmdCover.java
package me.ag2s.umdlib.domain;

import java.io.File;
import java.io.IOException;
import me.ag2s.umdlib.tool.UmdUtils;
import me.ag2s.umdlib.tool.WrapOutputStream;

public class UmdCover {
   private static int DEFAULT_COVER_WIDTH = 120;
   private static int DEFAULT_COVER_HEIGHT = 160;
   private byte[] coverData;

   public UmdCover() {
   }

   public UmdCover(byte[] coverData) {
      this.coverData = coverData;
   }

   public void load(File f) throws IOException {
      this.coverData = UmdUtils.readFile(f);
   }

   public void load(String fileName) throws IOException {
      this.load(new File(fileName));
   }

   public void initDefaultCover(String title) throws IOException {
   }

   public void buildCover(WrapOutputStream wos) throws IOException {
      if (this.coverData != null && this.coverData.length != 0) {
         wos.writeBytes(35, 130, 0, 1, 10, 1);
         byte[] rb = UmdUtils.genRandomBytes(4);
         wos.writeBytes(rb);
         wos.write(36);
         wos.writeBytes(rb);
         wos.writeInt(this.coverData.length + 9);
         wos.write(this.coverData);
      }
   }

   public byte[] getCoverData() {
      return this.coverData;
   }

   public void setCoverData(byte[] coverData) {
      this.coverData = coverData;
   }
}

File: src/main/kotlin/me/ag2s/umdlib/domain/UmdCover.kt
package me.ag2s.umdlib.domain

import java.io.File
import me.ag2s.umdlib.tool.UmdUtils
import me.ag2s.umdlib.tool.WrapOutputStream

class UmdCover(var coverData: ByteArray? = null) {
    fun load(file: File) { coverData = UmdUtils.readFile(file) }
    fun load(fileName: String) = load(File(fileName))
    fun initDefaultCover(title: String) = Unit
    fun buildCover(wos: WrapOutputStream) {
        val data = coverData ?: return
        if (data.isEmpty()) return
        wos.writeBytes(35, 130, 0, 1, 10, 1)
        val randomBytes = UmdUtils.genRandomBytes(4)
        wos.writeBytes(randomBytes)
        wos.write(36)
        wos.writeBytes(randomBytes)
        wos.writeInt(data.size + 9)
        wos.write(data)
    }
}
